src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1234,10 +1236,121 @@ TargetListOnPartitionColumn(Query *query, List *targetEntryList)
 		}
 	}
 
+	/*
+	 * We could stil behave as if the target list is on partition column if
+	 * all range table entries are reference tables and there is no column
+	 * references to any outer queries.
+	 */
```

Comment:
what if we have const expressions or functions ?

Reply:
We currently support columns only for hash distributed tables. Although it is possible to support those expressions on reference tables, I don't think they are very useful for now. Given that we'd want to backport this change, I'm inclined to keep the changes smaller.